feature/amazon/src/main/java/com/revenuecat/purchases/amazon/storeProductConversions.kt

```diff
@@ -1,25 +1,47 @@
 package com.revenuecat.purchases.amazon
 
-import com.amazon.device.iap.internal.model.ProductBuilder
+import android.os.Parcelable
 import com.amazon.device.iap.model.Product
+import com.revenuecat.purchases.ProductType
 import com.revenuecat.purchases.common.LogIntent
 import com.revenuecat.purchases.common.MICROS_MULTIPLIER
 import com.revenuecat.purchases.common.log
 import com.revenuecat.purchases.models.StoreProduct
+import com.revenuecat.purchases.models.ComparableData
+import com.revenuecat.purchases.parceler.JSONObjectParceler
+import kotlinx.parcelize.Parcelize
+import kotlinx.parcelize.TypeParceler
 import org.json.JSONObject
 import java.math.BigDecimal
 import java.util.regex.Pattern
-import com.amazon.device.iap.model.ProductType as AmazonProductType
 
-val StoreProduct.amazonProduct: Product
-    get() = ProductBuilder()
-        .setSku(originalJson.getString("sku"))
-        .setProductType(originalJson.getProductType("productType"))
-        .setDescription(originalJson.getString("description"))
-        .setPrice(originalJson.getString("price"))
-        .setSmallIconUrl(originalJson.getString("smallIconUrl"))
-        .setTitle(originalJson.getString("title"))
-        .setCoinsRewardAmount(originalJson.getInt("coinsRewardAmount")).build()
+@Parcelize
+@TypeParceler<JSONObject, JSONObjectParceler>()
+data class AmazonStoreProduct(
```

Comment:
Might be worth moving to its own `AmazonStoreProduct.kt` file, same as the bc4 version